File: android/src/main/java/ch/nadlo/oss/capacitor/user/defaults/UserDefaultPlugin.java
package ch.nadlo.oss.capacitor.user.defaults;

import android.content.Context;
import android.content.RestrictionsManager;
import android.os.Bundle;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "UserDefault")
public class UserDefaultPlugin extends Plugin {
    private final UserDefault implementation = new UserDefault();

    @Override
    protected void handleOnResume() {
        super.handleOnResume();

        RestrictionsManager myRestrictionsMgr =
                (RestrictionsManager) getContext().getSystemService(Context.RESTRICTIONS_SERVICE);

        Bundle appRestrictions = myRestrictionsMgr.getApplicationRestrictions();

        implementation.setAppRestrictions(appRestrictions);
    }

    @PluginMethod
    public void getByKey(PluginCall call) {
        String key = call.getString("key");

        JSObject ret = new JSObject();

        ret.put("value", implementation.readValue(key));
        ret.put("dict", implementation.appRestrictions);

        call.resolve(ret);
    }
}

File: ios/Plugin/UserDefaultPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(UserDefaultPlugin)
public class UserDefaultPlugin: CAPPlugin {
    private let implementation = UserDefault()
    
    @objc func getByKey(_ call: CAPPluginCall) {
        let key = call.getString("key") ?? ""
        
        call.resolve([
            "value": implementation.readValue(key: key)
        ])
    }
}
